infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/api/server/impls/KubernetesNamespaceMetaImpl.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.api.server.impls;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.eclipse.che.workspace.infrastructure.kubernetes.api.shared.KubernetesNamespaceMeta;
+
+/** @author Sergii Leshchenko */
+public class KubernetesNamespaceMetaImpl implements KubernetesNamespaceMeta {
+
+  private String name;
+  private Map<String, String> attributes;
+
+  public KubernetesNamespaceMetaImpl(String name) {
+    this.name = name;
+  }
+
+  public KubernetesNamespaceMetaImpl(String name, Map<String, String> attributes) {
+    this.name = name;
+    if (attributes != null) {
+      this.attributes = new HashMap<>(attributes);
+    }
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  @Override
+  public Map<String, String> getAttributes() {
+    if (attributes == null) {
+      attributes = new HashMap<>();
+    }
+    return attributes;
+  }
+
+  public void setAttributes(Map<String, String> attributes) {
+    this.attributes = attributes;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof KubernetesNamespaceMetaImpl)) {
```

Comment:
This is not a direct problem with this code, but we seem to be using `if (o != null && o.getClass() != getClass())` and the `instanceof` check interchangeably in various `*Impl`s. While almost the same, the semantics is slightly different (`getClass()` approach doesn't support subclassing nicely). What do we prefer?

Reply:
Personally I prefer to go with way as we use in most places: Workspace and k8s API use instanceof of other impls

Do you know some places where we use comparing `getClass()`?
I'm OK with `getClass` as well, but it must be solved in general but not in one places if instanceOf lead to issues.
I know that equals must be `Symmetric` and with instead of it's not BUT there was a big discussion previously and in the end we got nothing )

Are you OK with instanceOf in one more place?